### DISCUSSION TITLE
Date & Time Formats

### DISCUSSION THREAD

**Discussion by nofue** [sqlpage/SQLPage#616] [Q&A] [answered]
Is there a setting for the app to use the same date (and time) format everywhere?
In my form it displays as mm/dd/yyyy and hh:mm AM/PM, respectively. 

In a table and exported to CSV it reads as yyyy-mm-dd hh:mm:ss

I'd like to have it displayed as dd.mm.yyyy (hh:mm 24-hour style) everywhere. Is there a global setting I haven't seen yet?

**Comment by lovasoa** [comment, marked as answer]
Hello! 
  There is no such global setting in SQLPage. The way dates are displayed in forms depends on the user's browser's locale. It won't be displayed the same to someone in France or in the US. 
  SQLPage always exports to yyyy-mm-dd hh:mm:ss (which is standard, and better for interoperability). But you can always use your database's date and time functions to display dates in a format that you like more.

**Reply by nofue** [reply]
What is this setting in the shell component for? 
`  'de-AT'                     as language,`

**Reply by lovasoa** [reply]
This is not related to date or time formats. This sets the html lang attribute.

The `lang` attribute in HTML specifies the language of the document's content. It is used for:

1. **Accessibility**: Screen readers use it to select the correct pronunciation and accent when reading content.
2. **Search Engines**: Helps search engines understand the content's language, improving SEO and indexing accuracy.
3. **Spell Checking**: Browsers use it to apply appropriate spell-checking and grammar rules.
4. **Typography**: Some fonts or typographic conventions vary by language (e.g., hyphenation rules), and the `lang` attribute helps handle that.

**Reply by nofue** [reply]
Good to know, thanks.

**Comment by YunusMechery** [comment]
I didn't try but two hints:
https://www.postgresql.org/docs/current/runtime-config-client.html#GUC-DATESTYLE
and
https://github.com/lovasoa/SQLpage/blob/main/configuration.md#connection-initialization-scripts
